mhkit/wave/io/wave_hindcast.py

```diff
@@ -0,0 +1,84 @@
+import pandas as pd
+import numpy as np
+from rex import WaveX, MultiYearWaveX
+
+def read_US_wave_dataset(wave_path, parameter, lat_lon, years=None, tree=None, 
+                                 unscale=True, str_decode=True, hsds=True):
+    
+        """
+        Reads data from the WPTO wave hindcast data hosted on AWS. 
+
+        Note: To access the WPTO hindcast data, you will need to configure h5pyd for data access on HSDS. 
+        To get your own API key, visit https://developer.nrel.gov/signup/. 
+
+        To configure h5phd type 
+        hsconfigure
+        and enter at the prompt:
+        hs_endpoint = https://developer.nrel.gov/api/hsds
+        hs_username = None
+        hs_password = None
+        hs_api_key = {your key}
+        Parameters
+        ----------
+        wave_path : string
+            Path to US_Wave .h5 files
+            Available formats:
+                /nrel/US_wave/US_wave$_{year}.h5
+                /nrel/US_wave/US_wave$_*.h5 (Only use when specifying years parameter)
+        parameter: string
+            dataset parameter to be downloaded
+            spatial dataset options: directionality_coefficient, energy_period, maximum_energy_direction
+                mean_absolute_period, mean_zero-crossing_period, omni-directional_wave_power, peak_period
+                significant_wave_height, spectral_width, water_depth 
+        lat_lon: tuple or list of tuples
+            latitude longitude pairs at which to extract data 
+        years : list, optional
+            List of years to access, by default None
+        tree : str | cKDTree (optional)
+            cKDTree or path to .pkl file containing pre-computed tree
+            of lat, lon coordinates
+        unscale : bool (optional)
+            Boolean flag to automatically unscale variables on extraction
+        str_decode : bool (optional)
+            Boolean flag to decode the bytestring meta data into normal
+            strings. Setting this to False will speed up the meta data read.
+        hsds : bool (optional)
+            Boolean flag to use h5pyd to handle .h5 'files' hosted on AWS
+            behind HSDS
```

Comment:
Why would I want to use this?

Reply:
I think we can remove this. It would mostly be useful to people within the national labs (especially NREL) because they could use that flag to look for the data on Eagle (our super computer). Keeping it would make the function more friendly for NREL folks, but might confuse others.

Reply:
I chose to not remove this, but I did add a more descriptive docstring, let me know if more clarification is needed.